store/pruning/manager.go

```diff
@@ -177,83 +99,80 @@ func (m *Manager) SetSnapshotInterval(snapshotInterval uint64) {
 	m.snapshotInterval = snapshotInterval
 }
 
-// ShouldPruneAtHeight return true if the given height should be pruned, false otherwise
-func (m *Manager) ShouldPruneAtHeight(height int64) bool {
-	return m.opts.Interval > 0 && m.opts.GetPruningStrategy() != types.PruningNothing && height%int64(m.opts.Interval) == 0
-}
-
-// LoadPruningHeights loads the pruning heights from the database as a crash recovery.
-func (m *Manager) LoadPruningHeights(db dbm.DB) error {
+// GetPruningHeight returns the height which can prune upto if it is able to prune at the given height.
+func (m *Manager) GetPruningHeight(height int64) int64 {
 	if m.opts.GetPruningStrategy() == types.PruningNothing {
-		return nil
+		return 0
 	}
-	loadedPruneHeights, err := loadPruningHeights(db)
-	if err != nil {
-		return err
+	if m.opts.Interval <= 0 {
+		return 0
 	}
 
-	if len(loadedPruneHeights) > 0 {
-		m.pruneHeightsMx.Lock()
-		defer m.pruneHeightsMx.Unlock()
-		m.pruneHeights = loadedPruneHeights
+	if height%int64(m.opts.Interval) != 0 || height <= int64(m.opts.KeepRecent) {
+		return 0
 	}
 
-	loadedPruneSnapshotHeights, err := loadPruningSnapshotHeights(db)
-	if err != nil {
-		return err
+	// Consider the snapshot height
+	pruneHeight := height - 1 - int64(m.opts.KeepRecent) // we should keep the current height at least
```

Comment:
What if `pruneHeight` gets negative?

Reply:
this will be checked in `PruneStores` when pruning in rootmulti